keps/sig-node/2371-cri-pod-container-stats/README.md

```diff
@@ -0,0 +1,416 @@
+    string name = 1;
+    // Cumulative count of bytes received.
+    Uint64Value rx_bytes = 2;
+    // Cumulative count of receive errors encountered.
+    Uint64Value rx_errors = 2;
+    // Cumulative count of bytes transmitted.
+    Uint64Value tx_bytes = 2;
+    // Cumulative count of transmit errors encountered.
+    Uint64Value tx_errors = 2;
+}
+
+// ProcessStats are stats pertaining to processes.
+message ProcessStats {
+    // Number of processes
+    Uint64Value process_count = 1;
+}
+```
+
+#### Kubelet
+
+Once all required CRI changes are completed, Kubelet can update it's CRI stats provider to stop fetching metrics from cAdvisor and instead obtain the metrics from the CRI for container and pods.
+
+* TODO: How to handle migration to newly created "full" CRI stats provider?
+    * Option #1: Modify existing CRI stats provider by removing all usage of cAdvisor. May lead to backwards compatibility issue due to possible differences between CRI and cAdvisor metric collection
+    * Option #2: Create a new stats provider backed by a feature gate that provides backward compatibility and allows transition to fully CRI backed stats provider 
+
+Additional work may be required to evaluate other kubelet components (e.g. eviction, preemption, etc) that may be relying on container or pod level metrics. Ideally all components will rely on summary API thereby alleviating need for cAdvisor for container and pod level stats. This is also a requirement to be able to disable cAdvisor container metrics collection.
+
+#### cAdvisor
+
+Once CRI and Kubelet stats provider level changes are in place, we can evaluate disabling cAdvisor from collecting container and pod level stats. We may need to introduce new ability in cAdvisor to disable on-demand collection for certain cgroup hierarchies to ensure we can continue using cAdvisor for only node/machine level stats. 
+
+### User Stories [optional]
+
+#### Story 1
+As a cluster admin, I would like my node to be as performant as possible, and not waste resources by duplicating stats collection by CRI and cAdvisor.
+
+#### Story 2
+As a node maintainer, I would like to clarify the sources of truth for each stat the Kubelet reports to the Summary API, so that I can better find which component is to investigate for any issues.
+
+### A note on history
+There have been conversations in the past on [deprecating the stats Summary API][#core-proposal], in favor of allowing an on-demand model of stats gathering/reporting.
+This KEP does not attempt to implement that. These stats are largely considered [a part of Kubernetes][#keep-summary]. The stated goal of this KEP is to reduce the overhead
+on two entities reporting metrics, not totally changing what stats the Kubelet reports (with the exception of Accelerator and UserDefinedMetrics).
+
+Thus, this KEP largely the plan described [here][#plan], with some changes:
+
+- the `/metrics/cadvisor` endpoint will likely not be exposed
+	- **TODO** will this break users?
```

Comment:
Yes.  This will break users.  At a minimum, we should feature gate this, and have a slow deprecation with instructions on how to obtain the same behavior by running a cAdvisor daemonset.

Reply:
added a note about feature gating it